core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1543,10 +1535,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     preferredReplicaImbalanceCount.set(pric)
   }
 
-  private def triggerControllerMove(): Unit = {
-    activeControllerId.set(-1)
```

Comment:
Yes, I agree that perhaps we should improve the test in ControllerFailoverTest. My understanding is that the test tries to simulate a case that the controller gets into an illegal state and tries to see if the controller can recover from it. But the way that it simulates the illegal state is quite involved. Perhaps we can do the following. (1) Consolidate all try/catch illegalStateException to ControllerEventThread.doWork(). (2) Change the test by inserting a MockEvent type that throws illegalStateException.